RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpMetricCalculator.cs

```diff
@@ -77,10 +79,24 @@ public CaDETMemberMetrics CalculateMemberMetrics(MemberDeclarationSyntax member)
             return new CaDETMemberMetrics
             {
                 CYCLO = CalculateCyclomaticComplexity(member),
-                LOC = GetLinesOfCode(member.ToString())
+                LOC = GetLinesOfCode(member.ToString()),
+                NOP = GetNumberOfParameters(member),
+                NOLV = GetNumberOfLocalVariables(member)
             };
         }
 
+        private int GetNumberOfLocalVariables(MemberDeclarationSyntax method)
+        {
+            // TODO: Change this also to real value
+            return 10;
+        }
+
+        private int GetNumberOfParameters(MemberDeclarationSyntax method)
+        {
+            // First because of structure type of 'OfType<ParameterListSyntax>()'
+            return method.DescendantNodes().OfType<ParameterListSyntax>().First().Parameters.Count;
```

Comment:
- I find this way to get number of parameters
- There is some better i guess, so fell free to inform me about it

Reply:
This seems alright, but we should make a good suite of test data to see if the results of the calculation are as expected.

The main issue (the reason why other tests are failing) is that you do not handle the edge case with properties that lack parameter lists (even empty ones). Since we are calculating metrics for all types of executable methods (properties, constructors, methods) we need to check them as well.

Try replacing the body with this code:
            var paramLists = method.DescendantNodes().OfType<ParameterListSyntax>().ToList();
            return !paramLists.Any() ? 0 : paramLists.First().Parameters.Count;

Two things to consider:
- What is the situation when we have multiple parameter lists?
- Pro tip: Run all unit tests before committing anything :)